src/Analyzers/CSharp/CodeFixes/MakeFieldOrPropertyRequired/CSharpMakeFieldOrPropertyRequiredCodeFixProvider.cs

```diff
@@ -0,0 +1,135 @@
+        {
+            var setMethod = propertySymbol.SetMethod;
+
+            // Property must have a `set` or `init` accessor in order to be able to be required
+            if (setMethod is null)
+                return;
+
+            var containingTypeVisibility = propertySymbol.ContainingType.GetResultantVisibility();
+            var minimalAccessibility = (Accessibility)Math.Min((int)propertySymbol.DeclaredAccessibility, (int)setMethod.DeclaredAccessibility);
+
+            if (!CanBeAccessed(containingTypeVisibility, minimalAccessibility))
+                return;
+
+            RegisterCodeFix(context, CSharpCodeFixesResources.Make_property_required, nameof(CSharpCodeFixesResources.Make_property_required));
+        }
+        else if (fieldOrPropertySymbol is IFieldSymbol fieldSymbol)
+        {
+            var containingTypeVisibility = fieldSymbol.ContainingType.GetResultantVisibility();
+            var accessibility = fieldSymbol.DeclaredAccessibility;
+
+            if (!CanBeAccessed(containingTypeVisibility, accessibility))
+                return;
+
+            RegisterCodeFix(context, CSharpCodeFixesResources.Make_field_required, nameof(CSharpCodeFixesResources.Make_field_required));
+        }
+
+        static bool CanBeAccessed(SymbolVisibility containingTypeVisibility, Accessibility accessibility) => containingTypeVisibility switch
+        {
+            SymbolVisibility.Public => accessibility is Accessibility.Public,
+            SymbolVisibility.Internal => accessibility is >= Accessibility.Internal,
+            SymbolVisibility.Private => accessibility is >= Accessibility.Internal,
+            _ => throw ExceptionUtilities.Unreachable(),
+        };
+    }
+
+    protected override Task FixAllAsync(Document document, ImmutableArray<Diagnostic> diagnostics, SyntaxEditor editor, CodeActionOptionsProvider fallbackOptions, CancellationToken cancellationToken)
+    {
+        var root = editor.OriginalRoot;
+        var generator = editor.Generator;
+        var visitedFieldDeclarations = new HashSet<FieldDeclarationSyntax>();
+
+        foreach (var diagnostic in diagnostics)
+        {
+            var propertyDeclarationOrFieldVariableDeclarator = root.FindNode(diagnostic.Location.SourceSpan);
+
+            // If we are fixing field, do not apply new declaration modifiers just to variable declarator, but to the whole field declaration.
+            // This is observable when there are several variables in single filed declaration:
+            // `public string _myField, _myField1;` -> `public required string _myField, _myField1;`
+            // Without this branch the result would be:
+            // ```
+            // public required string _myField;
+            // public required string _myField2;
+            // ```
+            if (propertyDeclarationOrFieldVariableDeclarator is VariableDeclaratorSyntax { Parent.Parent: FieldDeclarationSyntax fieldDeclaration })
```

Comment:
need to check that thsi is what we have in the analyzer side.

Reply:
Sorry, what do you mean by that? The analyzer here is compiler since we are fixing a compiler warning. What do you want me to check then?